src/api/endpoints/roles.py

```diff
@@ -88,3 +88,71 @@ def delete(cls, role_query=None):
         role.delete()
         return response_builder(dict(status="success",
                                 message="Role deleted successfully."), 200)
+
+
+class SocietyRoleAPI(Resource):
+    """Contains functionality to change Society Executives."""
+
+    @token_required
+    @roles_required(["Success Ops"])
+    def put(self):
+        """Change the a society executives."""
+        payload = request.get_json(silent=True)
+        if not payload:
+            return response_builder(dict(
+                message="Executive for change must be provided"
+            ), 400)
+
+        if not payload.get("role") and payload.get("society") and payload.get(
+                                                                    "name"):
+            return response_builder(dict(
+                message="Role, society and individual for change "
+                        "must be provided"
+            ), 400)
+
+        society = Society.query.filter_by(name=payload.get("society")).first()
```

Comment:
By all means;
In line 113, I am finding the society object from which the exec is being changed.
In line 114, I am finding the role object that is being changed.
In line 116, I am finding the user objects that have the specified role related to them by querying(through the join method) the association table created in the models. Since the roles are not saved as a column in the User object, rather as an array, I was unable to find a method to query the user that belonged to both the society and had the role specified without looping.
As such, I decided to limit the looping to a constant of 4 objects (since there will always only be as many at the position as there are societies.)
The looping within 4 objects will be quite swift.

From line 121, the list of 4 objects with the role specified from before we find the object that belongs to the society specified earlier.

Then in line 127, we find the new exec primed to be promoted.